app/src/lib/stores/commit-status-store.ts

```diff
@@ -380,6 +417,154 @@ export class CommitStatusStore {
       }
     })
   }
+
+  /**
+   * Retrieve GitHub Actions workflows, jobs, and logs for the branch and apply
+   * logs to matching check run output.
+   */
+  public async getLatestPRWorkflowRunsLogsForCheckrun(
+    repository: GitHubRepository,
+    ref: string,
+    branchName: string,
+    checkRuns: ReadonlyArray<IRefCheck>
+  ): Promise<ReadonlyArray<IRefCheck>> {
+    const key = getCacheKeyForRepository(repository, ref)
+    const subscription = this.subscriptions.get(key)
+    if (subscription === undefined) {
+      return checkRuns
+    }
+
+    const { endpoint, owner, name } = subscription
+    const account = this.accounts.find(a => a.endpoint === endpoint)
+    if (account === undefined) {
+      return checkRuns
+    }
+
+    const api = API.fromAccount(account)
+    const latestWorkflowRuns = await this.getLatestPRWorkflowRuns(
+      api,
+      owner,
+      name,
+      branchName
+    )
+
+    if (latestWorkflowRuns.length === 0) {
+      return checkRuns
+    }
+
+    const logHash = new Map<string, JSZip>()
+    const mappedCheckRuns = new Array<IRefCheck>()
+    for (let i = 0; i < checkRuns.length; i++) {
+      const cr = checkRuns[i]
```

Comment:
Why not just a "for each"? I don't see `i` being used 🤔 
```suggestion
    for (const cr of checkRuns.length) {
```